### ISSUE TITLE
SCIM Filter containing AND starting with NOT does not parse correctly

### ISSUE THREAD

**Issue by patrickmcdougle-okta** [thomaspoignant/scim2-parse-filter#55]
`userType ne "Employee" and not (emails co "example.com" or emails co "example.org")` is a test case in parse.test.ts. I would expect that flipping the two sides of the and would result in the same AST. However, when supplying `not (emails co "example.com" or emails co "example.org") and userType ne "Employee"` the AST that is returned is incorrect.

I will start working on a bug fix, but I am leaving Okta at the end of the month and may not be able to drive this to completion before my last day.

**Comment by thomaspoignant** [comment]
oh good catch, I never tried in that order so I missed it.
Let me know if you need help to resolve the problem.

**Comment by thomaspoignant** [comment]
I've checked your test and I wonder what you think is wrong because for me it is working as expected.
With the example of your PR, in the existing test, we had the string
```
userType ne "Employee" and not (emails co "example.com" or emails co "example.org")
```
That produces this AST

```
 and(op("ne", "userType", "Employee"), {
        op: "not",
        filter: or(
          op("co", "emails", "example.com"),
          op("co", "emails", "example.org")
        )
      })
```

If you flip the 2 sides of the string request, in the result we also have the order change in the AST.
For example, if you use
```
not (emails co "example.com" or emails co "example.org") and userType ne "Employee"
```

The output will be 
```
and(
  {
    op: "not",
    filter: or(
        op("co", "emails", "example.com"),
        op("co", "emails", "example.org")
      )
  },
  op("ne", "userType", "Employee")
)
```

If I am understanding correctly what you say, you expect to have something different here?
You expect to have the same output for both?

**Comment by patrickmcdougle-okta** [comment]
You are absolutely right that I botched the expected value of the test. However, the actual result is NOT your expected value either. I just pushed up the corrected test case and we'll see that it still fails.

**Comment by thomaspoignant** [comment]
Oh ok I see, yes the test is still failing, I will check if I can see what goes wrong.

**Comment by shaneshen-okta** [comment]
I am not sure if you are still woking on this.

**Comment by thomaspoignant** [comment]
@shaneshen-okta I had no time to work on this yet, if you want to have a look feel free.

**Comment by shaneshen-okta** [comment]
@thomaspoignant I add some fix and tests, please have a look. thanks

**Comment by thomaspoignant** [comment]
The issue is fixed by the PR #89.
